### DISCUSSION TITLE
Unconsistant use of Enum and const when dealing with literals

### DISCUSSION THREAD

**Discussion by LucasLev** [fastapi/fastapi#9996] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from typing import Literal

app = FastAPI()


@app.get("/regular")
def i_want_a_fruit(fruit: Literal["apple", "peach"]):
    return fruit


@app.get("/tropical")
def i_want_a_tropical_fruit(tropical_fruit: Literal["pineapple"]):
    return tropical_fruit
```

### Description

- Open the browser to the swagger docs
- navigate to the tropical endpoint

Expected :
- Drop down menu to select the tropical fruit (as in the /regular endpoint)

Current : 
- Regular string field with no hint as to what the possible values are
![image](https://github.com/tiangolo/fastapi/assets/60597983/ae898ef5-2825-445c-8f47-1694e3707787)
### Operating System

Linux, Windows, macOS

### FastAPI Version

0.100.1

### Python Version

3.11.3

**Comment by LucasLev** [comment]
We can see from the openapi.json 
```json
        "/tropical": {
            "get": {
                "summary": "I Want A Tropical Fruit",
                "operationId": "i_want_a_tropical_fruit_tropical_get",
                "parameters": [
                    {
                        "name": "tropical_fruit",
                        "in": "query",
                        "required": true,
                        "schema": {
                            "const": "pineapple",
                            "title": "Tropical Fruit"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful Response",
                        "content": {
                            "application/json": {
                                "schema": {}
                            }
                        }
                    },
                    "422": {
                        "description": "Validation Error",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/HTTPValidationError"
                                }
                            }
                        }
                    }
                }
            }
        }
```
That a literal of length one is converted to a const. I think they should still remain a swagger enum.

**Reply by nikstuckenbrock** [reply]
I've got the same problem here. The main problem about the `const` is that it is not supported by [OpenAPI](https://swagger.io/docs/specification/data-models/keywords/#unsupported) by now. I'm not that deep into the sourcecode of FastAPI and Pydantic, but I would assume that FastAPI uses Pydantic to generate a json_schema for the OpenAPI specification. As mentioned in the Pydantic documentation they use the standard json_schema which supports `const`.

**Reply by sh-at-cs** [reply]
@nikstuckenbrock Just FYI: That documentation is for OpenAPI 3.0. `const` [is supported](https://github.com/OAI/OpenAPI-Specification/issues/1313#issuecomment-533637768) in OpenAPI 3.1.

**Comment by marcelkwaschny** [comment]
Here is a little workaround to recursively replace all occurences of the `const` keyword in the dict of the generated openapi spec with the `enum` keyword:

```python
import copy
import json

from fastapi import FastAPI
from fastapi.openapi.utils import get_openapi


api = FastAPI()


def walk_and_replace(node):
    for item in node.values():
        if isinstance(item, dict):
            if "const" in item:
                item["enum"] = [item["const"]]
                item["type"] = "string"
                del item["const"]
            walk_and_replace(item)


def custom_openapi():
    if api.openapi_schema:
        return api.openapi_schema

    openapi_schema = get_openapi(
        title=api.title,
        version=api.version,
        openapi_version="3.0.2",
        description=api.description,
        routes=api.routes,
    )

    walk_and_replace(openapi_schema)

    api.openapi_schema = openapi_schema
    return api.openapi_schema


api.openapi = custom_openapi
```
